### DISCUSSION TITLE
Twirp Support

### DISCUSSION THREAD

**Discussion by delaneyj** [ent/ent#1854] [Ideas]
I'm very interested in the approach you have taken in regards auto generating hooks into services.  I'm a heavy user of [Twirp](https://twitchtv.github.io/twirp/docs/intro.html) and would like to know what the lift would be to support it in ent.  It's a much simpler version of rpc than gRPC but.

1. Works both in HTTP 1.1  & 2
2. Can be access with both binary and JSON data making it easy to test
3. Has amazing support for client/servers in Typescript making it fast and usable from the browser with no proxies.

I saw you have `protoc-gen-entgrpc`.  Given Twirp only supports unary client->server requests my hope is it would be mostly removing code from the service generation.

**Comment by mycroftcanner** [comment]
We use twirp for all our app engine standard endpoint... 

You might need to append a command to generate.go for generating the twirp code since the proto files generated by ent are compatible it seems.
